clause/clause.go

```diff
@@ -1,42 +1,111 @@
 package clause
 
 import (
-	"errors"
+	"clause/operators"
+	"clause/utils"
 	"strings"
 )
 
-// ErrInvalidOperator describes an invalid operator error
-var ErrInvalidOperator = errors.New("Invalid operator")
+const (
+	queryParamSep     string = "__"
+	fieldnameValueSep string = "="
+	pairSeriesSep     string = "&"
+	placeholderSQL    string = "SELECT * FROM ___ WHERE "
+	placeholderAND    string = " AND "
+	emptyString       string = " "
+	queryParamStart   string = "?"
+)
+
+// GenerateSQLQuery generates the actual SQL query
+func GenerateSQLQuery(g map[int][]string) string {
+	var s string
+
+	/* dfs through the Tree
+	   notice a list of visited nodes isn't necessary here
+	   as the traversal is like a linear pattern
+	   the produced tree is always linear (no recursion)
+	*/
+	for i := 0; i < len(g); i++ {
+		v := g[i]
+
+		if i == 0 {
+			/* edge case for between and not between */
+			if v[1] == "between" || v[1] == "nbetween" {
+				s = placeholderSQL
+				var u string
+				u = v[0] + operators.GetOperator(v[1]) + v[2]
+				s += strings.ReplaceAll(u, ",", placeholderAND)
+				continue
+			}
+			s += placeholderSQL +
+				v[0] +
+				operators.GetOperator(v[1]) +
+				v[2]
+			continue
+		}
```

Comment:
That is the most important func and there a couple of issues though
In this implementation values are replaced as is, meaning that if I pass `username=loking` I'll get  in SQL `username = loking` which will unfortunately generate an SQL Error. The expected SQL result is `username = 'loking'`.
We should be  able to properly type qs values